src/libfuncs.rs

```diff
@@ -270,14 +274,14 @@ where
     'this: 'ctx,
 {
     #[doc(hidden)]
-    pub(crate) fn results(self) -> impl Iterator<Item = Vec<Value<'ctx, 'this>>> {
-        self.results.into_iter().enumerate().map(|(branch_idx, x)| {
+    pub(crate) fn results(self) -> impl Iterator<Item = NativeResult<Vec<Value<'ctx, 'this>>>> {
+        self.results.into_iter().enumerate().map(|(_, x)| {
             x.into_iter()
                 .enumerate()
-                .map(|(arg_idx, x)| {
-                    x.into_inner().unwrap_or_else(|| {
-                        panic!("Argument #{arg_idx} of branch {branch_idx} doesn't have a value.")
```

Comment:
i think you are not writing the arg_idx value into the string here

Reply:
Ow you are right, thats why it wasn't being use, changing it, thanks.

Reply:
It should be right now